vcd/resource_vcd_dnat.go

```diff
@@ -35,49 +38,46 @@ func resourceVcdDNAT() *schema.Resource {
 			"network_name": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
 			},
 			"network_type": &schema.Schema{
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "ext",
-				ForceNew:     true,
 				ValidateFunc: validation.StringInSlice([]string{"ext", "org"}, false),
 			},
 			"external_ip": &schema.Schema{
 				Type:     schema.TypeString,
 				Required: true,
-				ForceNew: true,
 			},
 
 			"port": &schema.Schema{
 				Type:     schema.TypeInt,
-				Required: true,
-				ForceNew: true,
+				Optional: true,
+				Computed: true,
```

Comment:
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html states:

> Computed is often used to represent values that are **not user configurable** or can not be known at time of terraform plan or apply

Why are we changing "port", "translated_port" and "icmp_sub_type" to Computed when they seem to be casual user-settable parameters?

Reply:
Cause api still returns value of you haven't set. Computed allows for user to set and be provided by external means.

Reply:
Interesting and I see more occurrences of this approach in the code base, but it looks like it conflicts with the quote from Terraform above. @Didainius @dataclouder what do you think?

Reply:
Hm. It's kind of mixed feeling from me. On one way if the user didn't set it - then he probably doesn't need it. On the other hand it wouldn't hurt him. I haven't yet played with installed binary. Does it complain about diff being different on `terraform plan`? If it doesn't then it probably shouldn't be bad.

Reply:
It is due the reason we read and set everything to state file - for user to see state uptodate. As so `terraform plan` will complain cause user haven't set, but state file has. In similar approach Org Vdc was configured and do things.

Reply:
Are you sure it does? I think acceptance tests would be failing (state not clean) if it was so. Need to validate it. The `terraform plan` after `terraform apply` should report 0 changes. Otherwise we'll get bugs reported, we got that for networks constantly updating IPs previously.

Reply:
Definition of computed:
// The fields below relate to diffs.
//
// If Computed is true, then the result of this value is computed
// (unless specified by config) on creation.

Reply:
Thank you. I got tricked by the doc page above. Your pasted description from Terraform's schema.go file is much more correct. "unless specified by config" is the key here.